src/dscanner/analysis/properly_documented_public_functions.d

```diff
@@ -187,36 +187,81 @@ private:
 		return comment;
 	}
 
-	void checkDdocParams(size_t line, size_t column, const Parameters params)
+	void checkDdocParams(size_t line, size_t column, const Parameters params,
+					     const TemplateParameters templateParameters = null)
 	{
-		import std.algorithm.searching : canFind;
+		import std.array : array;
+		import std.algorithm.searching : canFind, countUntil;
+		import std.algorithm.iteration : map;
+		import std.algorithm.mutation : remove;
+		import std.range : indexed, iota;
+
+		// convert templateParameters into a string[] for faster access
+		const(TemplateParameter)[] templateList;
+		if (const tp = templateParameters)
```

Comment:
`if (templateParameters && templateParameters.templateParameterList)`

Reply:
it would then be 

```
if (templateParameters && templateParameters.templateParameterList)
    templateList = templateParameters.templateParameterList.items;
```

imho not much nicer.